staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -641,13 +642,17 @@ func (cfgCtl *configController) startRequest(ctx context.Context, rd RequestDige
 				numQueues = plState.pl.Spec.Limited.LimitResponse.Queuing.Queues
 
 			}
+			var flowDistinguisher string
 			var hashValue uint64
 			if numQueues > 1 {
-				flowDistinguisher := computeFlowDistinguisher(rd, fs.Spec.DistinguisherMethod)
+				flowDistinguisher = computeFlowDistinguisher(rd, fs.Spec.DistinguisherMethod)
 				hashValue = hashFlowID(fs.Name, flowDistinguisher)
 			}
 			startWaitingTime = time.Now()
 			klog.V(7).Infof("startRequest(%#+v) => fsName=%q, distMethod=%#+v, plName=%q, numQueues=%d", rd, fs.Name, fs.Spec.DistinguisherMethod, plName, numQueues)
+			if fs.Spec.DistinguisherMethod != nil {
+				ctx = debug.WithFlowDistinguisher(ctx, string(fs.Spec.DistinguisherMethod.Type), flowDistinguisher)
```

Comment:
1. Let's pass the flow distinguisher in all cases.  The flow identifier is (flow schema name, flow distinguisher) in all cases, even whenn the second part is the empty string.

2. Passing the DistinguisherMethod.Type is not very valuable.  It is redundant with configuration, and not needed because the full flow identifier is at hand.

3. This is a more expensive and obscure way to pass this information than just adding it to the arguments of `plState.queues.StartRequest`.

Reply:
these public interfaces are boundaries between sub-systems. i was trying to avoid touching the implementation or interfaces as we're adding a set of exterior debugging helpers.. the arguments passed in the context will not be used by the system except for the debugging dumpers. they're indeed obscure but that's intended b/c these arguments are for-debug-only.

Reply:
While `plState.queues.StartRequest` is technically a public interface, it is on an alpha-level internal component.  I think we should not hesitate to make mods that we think are useful, even if they are only for instrumentation purposes.

Reply:
removed context-attached arguments